Packs/MicrosoftDefenderThreatIntelligence/Integrations/MicrosoftDefenderThreatIntelligence/MicrosoftDefenderThreatIntelligence.py

```diff
@@ -0,0 +1,556 @@
+        """
+        odata_query = "?"
+        odata_query += f"$top={limit}&"
+        if odata:
+            odata_query += odata
+        if not intel_profile_id:
+            response = self.ms_client.http_request(
+                method="GET",
+                url_suffix=f"v1.0//security/threatIntelligence/intelProfiles{odata_query}",
+            )
+            return response.get("value", [])
+
+        """NEED TO CHECK IT LIMIT IS VALID HERE"""
+        response = self.ms_client.http_request(
+            method="GET",
+            url_suffix=f"v1.0//security/threatIntelligence/intelProfiles/{intel_profile_id}{odata_query}",
+        )
+
+        return [response]
+
+    def profile_indicators_list(self, intel_profile_id: str, intel_profile_indicator_id: str, odata: str, limit: int) -> list:
+        """
+        Retrieve intelligence profile indicators.
+
+        Args:
+            intel_profile_id (str): The ID of the intelligence profile.
+            intel_profile_indicator_id (str): The ID of a specific intelligence profile indicator.
+            odata (str): OData query parameters for filtering and formatting the response.
+            limit (int): Maximum number of indicators to return.
+
+        Returns:
+            list: A list of intelligence profile indicators, each containing information such as
+                    ID, source, first/last seen timestamps, and associated artifacts.
+        """
+        odata_query = "?"
+        odata_query += f"$top={limit}&"
+        if odata:
+            odata_query += odata
+
+        if not intel_profile_indicator_id:
+            response = self.ms_client.http_request(
+                method="GET",
+                url_suffix=f"v1.0//security/threatIntelligence/intelProfiles/{intel_profile_id}/indicators{odata_query}",
+            )
+            return response.get("value", [])
+
+        """NEED TO CHECK IF LIMIT IS VALID HERE"""
+        response = self.ms_client.http_request(
+            method="GET",
+            url_suffix=f"v1.0//security/threatIntelligence/intelligenceProfileIndicators/{intel_profile_indicator_id}{odata_query}",
+        )
+        return [response]
+
+    def host(self, host_id: str, odata: str) -> dict:
+        """
+        Retrieve host information by host ID.
+
+        Args:
+            host_id (str): The ID of the host to retrieve information for.
+            odata (str): OData query parameters for filtering and formatting the response.
+
+        Returns:
+            dict: A dictionary containing host information including ID, first/last seen timestamps,
+                    registrar, and registrant details.
+        """
+        odata_query = "?"
+        if odata:
+            odata_query += odata
+
+        return self.ms_client.http_request(
+            method="GET",
+            url_suffix=f"v1.0//security/threatIntelligence/hosts/{host_id}{odata_query}",
```

Comment:
If the odata is empty than we will just add a ? after the {host_id} with no other values is this ok?